src/commands/git-metadata/git.ts

```diff
@@ -93,3 +111,16 @@ export const getCommitInfo = async (
 
   return new CommitInfo(hash, remote, trackedFiles)
 }
+
+export const getCommitInfoBasic = async (git: simpleGit.SimpleGit): Promise<CommitInfo> => {
```

Comment:
Could this function be factorized with `getCommitInfo`?
As the difference is whether the error is thrown of printed on the standard output, it could always be thrown and the display of the error to the stdout should be handled by the command itself.
The model we used in the synthetics command was to export as a library all functions that are "general" and implemented in `cli.ts` anything that is specific to the CLI usage/interaction of the command. With this model, handling the standard output of the function should be done in `cli.ts`.

Reply:
On the global model: In our case the cli specific logic is inside `upload.ts`. I agree that it'd be great to reduce code duplication if possible so that changes don't have to be applied twice in `upload.ts` and `library.ts`. As long as it's not a complete headache that is.

Reply:
I didn't want to touch the existing implementation because I wasn't sure if there were any specific reasons the code was written the way it was. I agree that it makes the most sense to me to add the base functionality to `library` or `git` or whatever else is exported, and leave the pretty-format and printing to the CLI command (`upload`). This way, we can just handle the errors in the functions that are calling the library/git commands.

 I didn't want to introduce breaking behavior and re-writing the code to throw an error instead of printing and returning undefined. If you'd like, I can go ahead and try to refactor this a little so that we're printing the error inside the calling code instead of inside the library. @jybp and @m-rousse, thoughts?

Reply:
I've gone ahead and refactored `getCommitInfo` to just throw the errors. The onus should be on the caller to catch and handle these appropriately.